### DISCUSSION TITLE
WP roles and EME membership

### DISCUSSION THREAD

**Discussion by f14m07** [liedekef/events-made-easy#182] [Q&A]
Hi Franky,
I am looking to combine a WP role with an EME membership. When an EME person would be added to a membership they would be assigned a specific WP role. When an EME person would be removed from the same membership they would have the WP role removed.
Is this possible today? Or should a php function be written? 
I saw on the documentation that there is an existing hook for inserting a member but I did not see one for removing?
Thanks for the support.
Frédéric

**Comment by liedekef** [comment]
EME can check on EME memberships for page access, so no need for roles on its own. Anyway, this change will implement an action just before a member gets removed:
https://github.com/liedekef/events-made-easy/commit/01c41caa9cabf46c79d3bc23a223abcb3d748649

**Comment by f14m07** [comment]
I use the WP role to display the menus. Depending on the WP role the menus displayed are not the same.
Thanks anyway for your quick answer and the new hook

**Comment by f14m07** [comment]
I have created my function for the eme_insert_member_action hook. The problem is that it works even if the membership is not active, i.e. before the payment or if the payment is not made. 
Would there be a hook when the membership is activated and another one when it is deactivated?

**Comment by liedekef** [comment]
You can check if a member is active using:
` eme_is_active_member($member)`
(which returns 1 if active)
There is no "deactivation", just delete for a member. And that delete-action hook only fires on deletion.

**Reply by liedekef** [reply]
Btw: that won't help with what you want I think. In your case, look at eme_member_email_action ==> that action fires when a status is changed for a member (and an email is desired). You can abuse it :-)

**Comment by f14m07** [comment]
I understand but at the time of the hook the membership is not yet active, and when it is activated I have no hook. I will look at eme_member_email_action

**Reply by liedekef** [reply]
That is why I said to tr to use eme_member_email_action

**Comment by liedekef** [comment]
Also, I added a new hook: eme_member_status_change_action : https://github.com/liedekef/events-made-easy/commit/47ad33ddd3544d4861c70e783f15934700db6bba
(executed after the member status changes). 2 params: $member_id and status

**Comment by f14m07** [comment]
Here are the functions I have written, to help someone if necessary

```php
add_action('eme_insert_member_action', 'ajoutrolecarte') ;
function ajoutrolecarte($member){
	$personne_id=$member['person_id'];
	$personne=eme_get_person($personne_id);	
	if($member['membership_id']==2){$user = get_user_by( 'ID', $personne['wp_id'] );$user->add_role( 'cartemembre' );}
 }

add_action('eme_delete_member_action', 'enleverolecarte') ;
function enleverolecarte($member){
	$personne_id=$member['person_id'];
	$personne=eme_get_person($personne_id);	
	if($member['membership_id']==2){$user = get_user_by( 'ID', $personne['wp_id'] );$user->remove_role( 'cartemembre' );}
}```
